tests/unit/test_repo.py

```diff
@@ -146,3 +151,292 @@ def test_can_add_repositories_from_string_with_options(self):
+                repository_mapping = apt.RepositoryMapping()
+        assert not repository_mapping._repository_map
+
+        with patch(
+            "builtins.open", new_callable=mock_open, read_data=ubuntu_sources_deb822_with_comments
+        ):
+            with patch.object(apt.logger, "debug") as debug:
+                repository_mapping.load_deb822("FILENAME")
+        assert sorted(repository_mapping._repository_map.keys()) == [
+            "deb-http://nz.archive.ubuntu.com/ubuntu/-noble",
+            "deb-http://nz.archive.ubuntu.com/ubuntu/-noble-backports",
+            "deb-http://nz.archive.ubuntu.com/ubuntu/-noble-updates",
+        ]
+        debug.assert_called_once_with(
+            ANY,
+            1,  # number of errors
+            "Missing key 'Types' for entry starting on line 11 in FILENAME.",
+        )
+
+    def test_init_with_deb822(self):
+        """Mock file opening to initialise a RepositoryMapping from deb822 and one-line-style.
+
+        They should be equivalent with the sample data being used.
+        """
+
+        def isnt_file(f: str) -> bool:
+            return False
+
+        def iglob_list(s: str) -> typing.Iterable[str]:
+            if s.endswith(".list"):
+                return ["FILENAME"]
+            return []
+
+        def iglob_sources(s: str) -> typing.Iterable[str]:
+            if s.endswith(".sources"):
+                return ["FILENAME"]
+            return []
+
+        with patch("builtins.open", new_callable=mock_open, read_data=ubuntu_sources_one_line):
+            with patch("os.path.isfile", new=isnt_file):
+                with patch("glob.iglob", new=iglob_list):
+                    repository_mapping_list = apt.RepositoryMapping()
+
+        with patch("builtins.open", new_callable=mock_open, read_data=ubuntu_sources_deb822):
+            with patch("os.path.isfile", new=isnt_file):
+                with patch("glob.iglob", new=iglob_sources):
+                    repository_mapping_sources = apt.RepositoryMapping()
+
+        list_keys = sorted(repository_mapping_list._repository_map.keys())
+        sources_keys = sorted(repository_mapping_sources._repository_map.keys())
+        assert sources_keys == list_keys
+
+        for list_key, sources_key in zip(list_keys, sources_keys):
+            list_repo = repository_mapping_list[list_key]
+            sources_repo = repository_mapping_sources[sources_key]
+            assert list_repo.enabled == sources_repo.enabled
+            assert list_repo.repotype == sources_repo.repotype
+            assert list_repo.uri == sources_repo.uri
+            assert list_repo.release == sources_repo.release
+            assert list_repo.groups == sources_repo.groups
+            assert list_repo.gpg_key == sources_repo.gpg_key
+            assert (
+                list_repo.options  # pyright: ignore[reportUnknownMemberType]
+                == sources_repo.options  # pyright: ignore[reportUnknownMemberType]
+            )
```

Comment:
Unit test coverage of parsing deb822 is probably sufficient imo, but I'd like to add unit tests (or cases) for other operations touched by this PR (add, disable, gpg_key), as well as integration tests.

Reply:
Yeah, a couple of integration tests would be good.